internal/transport/http_util.go

```diff
@@ -120,8 +118,32 @@ type decodeState struct {
 	statsTags      []byte
 	statsTrace     []byte
 	contentSubtype string
+}
+
+func newDecodeState(serverSide bool, ignoreContentType bool) *decodeState {
+	return &decodeState{
+		serverSide:        serverSide,
+		ignoreContentType: ignoreContentType,
+	}
+}
+
+// Records the states during HPACK decoding. Must be reset once the
+// decoding of the entire headers are finished.
+type decodeState struct {
 	// whether decoding on server side or not
 	serverSide bool
+	// ignoreContentType indicates whether when processing the HEADERS frame, ignoring checking the
+	// content-type is grpc or not.
+	//
+	// If we've already received a HEADERS frame which indicates gRPC peer, then we can ignore
+	// content-type for the following HEADERS frame (i.e. Trailers).
```

Comment:
Trailers (after headers) shouldn't even have a content-type - or if it does, we should always ignore it, no?

Reply:
rephrased. Mentioned Trailers (after headers) shouldn't even have a content-type, and we ignore checking it.